pkg/nsx/services/staticroute/staticroute.go

```diff
@@ -146,16 +148,21 @@ func (service *StaticRouteService) ListStaticRoute() []*model.StaticRoutes {
 	return staticRouteSet
 }
 
-func (service *StaticRouteService) Cleanup() error {
+func (service *StaticRouteService) Cleanup(ctx context.Context) error {
 	staticRouteSet := service.ListStaticRoute()
 	log.Info("cleanup staticroute", "count", len(staticRouteSet))
 	for _, staticRoute := range staticRouteSet {
 		path := strings.Split(*staticRoute.Path, "/")
 		log.Info("removing staticroute", "staticroute path", *staticRoute.Path)
-		err := service.DeleteStaticRouteByPath(path[2], path[4], path[6], *staticRoute.Id)
-		if err != nil {
-			log.Error(err, "remove staticroute failed", "staticroute id", *staticRoute.Id)
-			return err
+		select {
+		case <-ctx.Done():
+			return nsxutil.TimeoutFailed
```

Comment:
To be robust, we can probably enhance this with even stronger error types - e.g. `TimeoutDeletingStaticRoute`, but I don't know how valuable going this granular is. Not a blocker, for sure, just a thought.